mod/async/pkg/broker/broker.go

```diff
@@ -35,6 +38,10 @@ type Broker[T any] struct {
 	msgs chan T
 	// timeout is the timeout for sending a msg to a client.
 	timeout time.Duration
+	// mu is the mutex for the clients map.
+	mu sync.Mutex
```

Comment:
I believe dev had the mutex originally in the first iteration, and then removed it as it wasn't necessary